src/components/LenderProfile/LenderMap.vue

```diff
@@ -0,0 +1,199 @@
+<template>
+	<section>
+		<h4 class="data-hj-suppress tw-mb-2">
+			{{ lenderMapTitle }}
+		</h4>
+		<kv-map
+			class="
+				tw-rounded
+				tw-overflow-hidden
+				tw-mb-3
+			"
+			:auto-zoom-delay="500"
+			:aspect-ratio="1.8"
+			:lat="30"
+			:long="1"
+			:zoom-level="2"
+			:use-leaflet="true"
+			:show-zoom-control="true"
+			:allow-dragging="true"
+			:show-labels="false"
+			:countries-data="countriesData"
+			:show-fundraising-loans="showFundraisingLoans"
+		/>
+		<div>
+			<p class="tw-mb-1 tw-font-medium">
+				Legend:
+			</p>
+			<div class="tw-grid tw-grid-cols-2 md:tw-grid-cols-3 tw-gap-2">
+				<div
+					v-for="legend in legendData"
+					:key="legend.color"
+					class="tw-flex tw-items-start tw-gap-1"
+				>
+					<span
+						:class="[
+							'tw-w-4',
+							'tw-h-2',
+							'tw-border',
+							legend.color,
+						]"
+					></span>
+					<p class="tw-text-small tw-font-medium">
+						{{ legend.label }}
+					</p>
+				</div>
+			</div>
+			<div class="tw-flex tw-items-center tw-gap-1 tw-mt-2">
+				<span
+					class="tw-w-1 tw-h-1 tw-border tw-rounded-full tw-bg-brand-900"
+				></span>
+				<p class="tw-text-small tw-font-medium">
+					Fundraising loans
+				</p>
+			</div>
+		</div>
+		<div class="tw-mt-3">
+			<p class="tw-mb-1 tw-font-medium">
+				Options:
+			</p>
+			<kv-checkbox
+				class="tw-text-small tw-font-medium custom-checkbox"
+				v-model="showFundraisingLoans"
+			>
+				Show fundraising loans
+			</kv-checkbox>
+		</div>
+	</section>
+</template>
+
+<script>
+import numeral from 'numeral';
+import KvMap from '@/components/Kv/KvMap';
+import KvCheckbox from '~/@kiva/kv-components/vue/KvCheckbox';
+
+export const getIntervals = (min, max, nbIntervals) => {
+	const size = Math.floor((max - min) / nbIntervals);
+	const result = [];
+
+	if (size <= 0) return [[min, max]];
+
+	for (let i = 0; i < nbIntervals; i += 1) {
+		let inf = min + (i * size);
+		let sup = ((inf + size) < max) ? inf + size : max;
+
+		if (i > 0) {
+			inf += (1 * i);
+			sup += (1 * i);
+		}
+
+		if (i > 0 && sup > max) {
+			sup = max;
+		}
+
+		if (i === (nbIntervals - 1)) {
+			if (sup < max || sup > max) sup = max;
+		}
+
+		result.push([inf, sup]);
+
+		if (sup >= max) break;
+	}
+
+	return result;
+};
+
+export const mapColors = [
+	100,
+	300,
+	500,
+	650,
+	800,
+	1000,
+];
+
+export default {
+	name: 'LenderMap',
+	components: {
+		KvMap,
+		KvCheckbox,
+	},
+	props: {
+		lenderInfo: {
+			type: Object,
+			required: true,
+		},
+	},
+	data() {
+		return {
+			showFundraisingLoans: true,
+		};
+	},
+	computed: {
+		lenderMapTitle() {
+			return this.lenderInfo?.name
+				? `${this.lenderInfo.name}'s Lending Activity by Country`
+				: 'Lending Activity by Country';
+		},
+		countriesData() {
+			return (this.lenderInfo?.statsPerCountry?.values ?? []).map(stat => ({
+				label: stat.country?.name ?? '',
+				value: stat.loanCount ?? '',
+				lat: stat.country?.geocode?.latitude ?? 0,
```

Comment:
Are we generally guaranteed the the stat data exists or should we be filtering the array before we map the data?

Reply:
There are some cases for private/anonymous returning null, when we are returning an empty array.
In other cases we are guaranteed that stat data exists.